File: Memo01/app/src/main/java/com/callor/memo/db/Injection.java
package com.callor.memo.db;

import android.content.Context;

public class Injection {


        public static MemoDataSource provideUserDataSource(Context context) {
            MemoDB database = MemoDB.getInstance(context);
            return new MemoDataSourceImpl(database.getMemoDao());
        }

        public static ViewModelFactory provideViewModelFactory(Context context) {
            MemoDataSource dataSource = provideUserDataSource(context);
            return new ViewModelFactory(dataSource);
        }

}

File: MemoKt01/app/src/main/java/com/callor/memo/db/Injection.kt
package com.callor.memo.db

import android.content.Context

object Injection {


    fun provideUserDataSource(context: Context): MemoDataSource {
        val database = MemoDB.getInstance(context)
        return MemoDataSourceImpl(database!!.memoDao)
    }

    fun provideViewModelFactory(context: Context): ViewModelFactory {
        val dataSource = provideUserDataSource(context)
        return ViewModelFactory(dataSource)
    }

}
